src/pruner_util.cpp

```diff
@@ -91,6 +91,55 @@ int get_exit_code(cstring name, P4PRUNER::PrunerConfig pruner_conf) {
     return WEXITSTATUS(system(command.c_str()));
 }
 
+cstring get_error_string(cstring name, P4PRUNER::PrunerConfig pruner_conf) {
+    cstring command = pruner_conf.compiler;
+    command += " ";
+    command += name;
+    // Apparently popen doesn't like stderr hence redirecting stderr to stdout
+    command += " 2>&1";
+
+    char buffer[128];
+    cstring result = "";
+    FILE *pipe = popen(command, "r");
+
+    try {
+        while (fgets(buffer, sizeof buffer, pipe) != NULL) {
+            result += buffer;
+        }
+    } catch (...) {
+        pclose(pipe);
+        throw;
+    }
+    pclose(pipe);
+    return result;
+}
+
+bool is_crash_bug(cstring name, P4PRUNER::PrunerConfig pruner_conf) {
+    INFO("Checking if this is a crash bug");
+
+    cstring command = pruner_conf.compiler;
+    command += " ";
+    command += name;
+    command += " 2>\\dev\\null";
+    int out = WEXITSTATUS(system(command));
```

Comment:
Running the compiler twice seems unnecessary. Why is not possible to get the return code and output at once? Also: You can run this with the "--Wdisable " flag then no warnings will be printed.

Reply:
I felt that the individual functions might be required later seperately hence I kept the functions separate. Thanks for the flag.

Reply:
This is fairly inefficient. It is fine if we get both at once, then check the exit code first and the err msg later.
Just like this tree ha:
```
        exit_code
            |
        _________
        |       |
        1       20
        |
    _________
    |       |
    error   bug
```
All of this can be abstracted away in the `check_pruned_program` and `get_exit_code` functions